### PR TITLE
Monitor command

### PR DESCRIPTION
**Pull request by RoeyPrat**
This is a rebase of  https://github.com/andymccurdy/redis-py/pull/194 by @dougk7.
He implemented the MONITOR command in a similar way to PubSub, using a generator to get the messages.
I also fixed some PEP8 and formatting issues and added the command to README.
This issue was also mentioned in #805.
Thanks!

### CODE REVIEW

**Comment by RoeyPrat** [comment]
@andymccurdy I changed monitor to use context manager. also, the message is parsed.

File: redis/client.py [hunk 1]
```diff
@@ -2817,6 +2821,49 @@ def zadd(self, name, *args, **kwargs):
...
+    next_command() method returns one command from monitor
+    listen() method yields commands from monitor.
+    """
+    def __init__(self, connection_pool):
+        self.connection_pool = connection_pool
+        self.connection = self.connection_pool.get_connection('MONITOR')
+
+    def __enter__(self):
+        self.connection.send_command('MONITOR')
+        assert self.next_command()
```
[the remark this answers is not in the document]
Comment (RoeyPrat) [reply redis/client.py]: ok. I'm setting error to `RedisError('MONITOR failed: %s' % response)`

File: redis/client.py [hunk 2]
```diff
@@ -2817,6 +2821,49 @@ def zadd(self, name, *args, **kwargs):
...
+        self.connection.disconnect()
+        self.connection_pool.release(self.connection)
+
+    def next_command(self):
+        "Parse the response from a monitor command"
+        response = self.connection.read_response()
+        # This part with the OK is only needed when we connect initially.
+        if bool_ok(response):
+            return True
+        command_time, command_data = response.split(' ', 1)
```
[the remark this answers is not in the document]
Comment (RoeyPrat) [reply redis/client.py]: for now, I'm setting "If decode_responses=False, return monitor output without parsing".

**Comment by RoeyPrat** [comment]
I still need to create tests for this feature...

**Comment by andymccurdy** [comment]
@RoeyPrat I took a look at this a bit yesterday and found some issues.

1. When inspecting the response, it looks like the `command` key is set to just the first word of the command name. For instance, when another client calls `GET foo`, the `command` key within the monitor dict is simply `('GET',)`. `command` should either contain the full command, including all the args, or we should have a separate field called `command_name` that is just `GET`. And whatever we choose, this should just be a string, not a tuple.

2. As mentioned in a comment above, some command names are two words. The current regex only parses the first word and leaves the rest behind.

3. I think the `db` field should be an int rather than a string. We represent it as an int everywhere else in the client.

4. We're not taking any steps to verify that the response we're getting back from `next_command` corresponds to the sample command being issued in the test (in your test case, `PING`). In fact, the response from `next_command()` in your test actually describes a `SELECT` command which the connection is running automatically just after the socket is connected. If the test database is shared with other users or processes, we could actually see activity from those as well.

5. I discovered the above issue while trying to write a test that shows how the current implementation fails with certain encoding issues. I'm trying to get around the test limitations now and will update this when I do.

**Comment by andymccurdy** [comment]
@RoeyPrat I took your PR and made some changes attempting to fix the problems I outlined above. The branch is here: https://github.com/andymccurdy/redis-py/tree/monitor

I think everything works as expected with this. Can you review and let me know what you think? Thanks!

**Comment by RoeyPrat** [comment]
@andymccurdy - looks great! I see no problem in your branch.

**Comment by andymccurdy** [comment]
Merged. Thanks @RoeyPrat

**Pull request merged**